code/modules/projectiles/guns/ego_gun/waw.dm

```diff
@@ -54,6 +54,16 @@
 		return "<span class='notice'>Its bullets deal [chambered.BB.damage] randomly chosen damage.</span>"
 	return
 
+/obj/item/gun/ego_gun/hatred/attackby(obj/item/I, mob/living/user, params)
```

Comment:
Again, this may just benefit from the nihil variant not being a subtype and just being it's own one with it's bits redefined.